Desktop_Interface/isobuffer.cpp

```diff
@@ -132,57 +145,41 @@ void isoBuffer::writeBuffer_short(short* data, int len)
 	writeBuffer(data, len, 2048, [](short item) -> short {return item >> 4;});
 }
 
-short* isoBuffer::readBuffer(double sampleWindow, int numSamples, bool singleBit, double delayOffset)
+std::unique_ptr<short[]> isoBuffer::readBuffer(double sampleWindow, int numSamples, bool singleBit, double delayOffset)
 {
-	/* Refactor Note:
-	 *
-	 * Refactoring this function took a few passes were i made some assumptions:
-	 *  - round() should be replaced by floor() where it was used
-	 *  - int(floor(x)) and int(x) are equivalent (since we are always positive)
-	 *  - free(NULL) is a no-op. This is mandated by the C standard, and virtually all
-	 * implementations comply. A few known exceptions are:
-	 *    - PalmOS
-	 *    - 3BSD
-	 *    - UNIX 7
-	 *   I do not know of any non-compliant somewhat modern implementations.
+	/*
+	 * The expected behavior is to run backwards over the buffer with a stride
+	 * of timeBetweenSamples steps, and push the touched elements into readData.
+	 * If more elements are requested than how many are stored (1), the buffer
+	 * will be populated only partially. Modifying this function to return null
+	 * or a zero-filled buffer insted should be simple enough.
```

Comment:
insted -> instead

Reply:
Ahhh, the typos.  Maybe i should start running my changes through a spell checker. hahaha